zerver/lib/subscription_info.py

```diff
@@ -111,6 +113,7 @@ def build_unsubscribed_sub_from_stream_dict(
     raw_stream_dict = RawStreamDict(
         can_remove_subscribers_group_id=stream_dict["can_remove_subscribers_group"],
         date_created=timestamp_to_datetime(stream_dict["date_created"]),
+        deactivated=Stream.objects.get(id=stream_dict["stream_id"]).deactivated,
```

Comment:
Yikes! This will do a database query for each stream processed this way. Isn't the value already in the `stream_dict`? This seems like a potentially significant performance bug.

Reply:
That's right. Fixed it.